### PR TITLE
pscanrules: add example alert to Cache Control Scan Rule

### PR DESCRIPTION
**Pull request by thgomes**
Part of https://github.com/zaproxy/zaproxy/issues/6119

Changes:
- Add example alert functionality.
- Add unit tests for example alerts functionality.
- Add note to changelog.

### CODE REVIEW

File: addOns/pscanrules/src/main/java/org/zaproxy/zap/extension/pscanrules/CacheControlScanRule.java [hunk 1]
```diff
@@ -67,18 +67,18 @@ public void scanHttpResponseReceive(HttpMessage msg, int id, Source source) {
                     cacheControlHeaders.indexOf("no-store") < 0
                     || cacheControlHeaders.indexOf("no-cache") < 0
                     || cacheControlHeaders.indexOf("must-revalidate") < 0) {
-                this.raiseAlert(HttpHeader.CACHE_CONTROL, cacheControlHeaders);
+                this.createAlert(HttpHeader.CACHE_CONTROL, cacheControlHeaders).raise();
```
Comment (thc202) [inline addOns/pscanrules/src/main/java/org/zaproxy/zap/extension/pscanrules/CacheControlScanRule.java:70]: Extract a constant for the header and use it the 2 required places.

File: addOns/pscanrules/CHANGELOG.md [hunk 2]
```diff
@@ -6,6 +6,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 ## Unreleased
 ### Added
 - The X-Backend-Server Scan Rule now includes example alert functionality for documentation generation purposes (Issue 6119).
+- The Cache Control Scan Rule now includes example alert functionality for documentation generation purposes (Issue 6119).
```
Comment (thc202) [inline addOns/pscanrules/CHANGELOG.md:9]: See https://github.com/zaproxy/zap-extensions/pull/4705#discussion_r1246668143

Comment (thc202) [reply addOns/pscanrules/CHANGELOG.md:9]: The other PR was merged in the meantime, this should just need to add to the existing entry.

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

**Review by kingthorin — APPROVED** [review]

**Pull request merged**